Scripts/Editor/Utils/CodeGenerationUtility.cs

```diff
@@ -203,11 +212,25 @@ public static void GenerateStaticCollectionScript(ScriptableObjectCollection col
                 directives.Add(collection.GetType().Namespace);
                 directives.Add(typeof(List<>).Namespace);
                 directives.Add("System.Linq");
+                directives.Add("System");
                 directives.AddRange(GetCollectionDirectives(collection));
                 string className = collection.GetItemType().Name;
+
+                if (className.Contains(ReferenceConverter.BaseClassNamePostFix) && collection.Items.Count(item => item.IsReference()) > 0)
+                {
+                    int baseClassPostFixLength = ReferenceConverter.BaseClassNamePostFix.Length;
+                    if (className.Length > baseClassPostFixLength)
+                        className = className.Remove(className.Length - baseClassPostFixLength, baseClassPostFixLength);
+                }
                 if (!writeAsPartial)
                     className = fileName;
```

Comment:
Missing `{ }`
I try to keep constant, if the `if` or the `else` has brackets, both need